solr/core/src/java/org/apache/solr/pkg/PackageLoader.java

```diff
@@ -274,20 +349,25 @@ public void writeMap(EntryWriter ew) throws IOException {
         version.writeMap(ew);
       }
 
-      Version(Package parent, PackageAPI.PkgVersion v) {
+      Version(Package parent, PackageAPI.PkgVersion v, Path localPkgDir) {
         this.parent = parent;
         this.version = v;
         List<Path> paths = new ArrayList<>();
-
-        List<String> errs = new ArrayList<>();
-        coreContainer.getPackageStoreAPI().validateFiles(version.files, true, s -> errs.add(s));
-        if(!errs.isEmpty()) {
-          throw new RuntimeException("Cannot load package: " +errs);
-        }
-        for (String file : version.files) {
-          paths.add(coreContainer.getPackageStoreAPI().getPackageStore().getRealpath(file));
+        if(localPkgDir != null) {
+          for (String file : v.files) {
+            if(file.charAt(0)== '/') file =file.substring(1);
+            paths.add( localPkgDir.resolve(file).toAbsolutePath()) ;
+          }
+        } else {
```

Comment:
Some code smell here with if-then-else based on quite different behaviour - local packages that are just added t path vs remote packages that need validation etc. Can we have sublcassing instead, either `ValidatingPackageLoader` and `LocalPackageLoader`, and the same for Package.Version?

Reply:
Yeah, I think a subclass with a different name would be better